File: examples-sdk/zig/010-basic-websocket-chat/main.zig
// 010-basic-websocket-chat — Zig SDK equivalent
// Compile: vil compile --from zig --input 010-basic-websocket-chat/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("websocket-chat", 8080);
    var chat = vil.Service.init("chat");
    chat.endpoint("GET", "/", "index");
    chat.endpoint("GET", "/ws", "ws_handler");
    chat.endpoint("GET", "/stats", "stats");
    server.service(&chat);
    server.compile();
}

File: examples-sdk/go/010-basic-websocket-chat/main.go
// 010 — WebSocket Chat
// Equivalent to: examples/010-basic-websocket-chat (Rust)
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	server := vil.NewServer("websocket-chat", 8080)

	chat := vil.NewService("chat")
	chat.Endpoint("GET", "/", "index")
	chat.Endpoint("GET", "/ws", "ws_handler")
	chat.Endpoint("GET", "/stats", "stats")
	server.Service(chat)

	server.Compile()
}
